qa/common/gen_ensemble_model_utils.py

```diff
@@ -0,0 +1,428 @@
+
+        config_dir = models_dir + "/" + model_name
+        config = '''
+name: "{}"
+platform: "ensemble"
+max_batch_size: {}
+version_policy: {}
+input [
+  {{
+    name: "INPUT0"
+    data_type: {}
+    dims: [ {} ]
+  }},
+  {{
+    name: "INPUT1"
+    data_type: {}
+    dims: [ {} ]
+  }}
+]
+output [
+  {{
+    name: "OUTPUT0"
+    data_type: {}
+    dims: [ {} ]
+    label_filename: "output0_labels.txt"
+  }},
+  {{
+    name: "OUTPUT1"
+    data_type: {}
+    dims: [ {} ]
+  }}
+]
+{}
+'''.format(model_name, max_batch, version_policy_str,
+            input_model_dtype, tu.shape_to_dims_str(input_shape),
+            input_model_dtype, tu.shape_to_dims_str(input_shape),
+            output0_model_dtype, tu.shape_to_dims_str(output0_shape),
+            output1_model_dtype, tu.shape_to_dims_str(output1_shape),
+            ensemble_schedule)
+
+        try:
+            os.makedirs(config_dir)
+        except OSError as ex:
+            pass # ignore existing dir
+
+        with open(config_dir + "/config.pbtxt", "w") as cfile:
+            cfile.write(config)
+
+        with open(config_dir + "/output0_labels.txt", "w") as lfile:
+            for l in range(output0_label_cnt):
+                lfile.write("label" + str(l) + "\n")
+
+def create_nop_modelconfig(models_dir, tensor_shape, tensor_dtype):
+    model_name = "nop_{}_{}".format(tensor_dtype, len(tensor_shape))
+    config_dir = models_dir + "/" + model_name
+    config = '''
+name: "nop_{dtype}_{dim_len}"
+platform: "custom"
+max_batch_size: {batch_size}
+default_model_filename: "libidentity.so"
+input [
+  {{
+    name: "INPUT0"
+    data_type: {dtype}
+    dims: [ {dim} ]
+  }},
+  {{
+    name: "INPUT1"
+    data_type: {dtype}
+    dims: [ {dim} ]
+  }}
+]
+output [
+  {{
+    name: "OUTPUT0"
+    data_type: {dtype}
+    dims: [ {dim} ]
+  }},
+  {{
+    name: "OUTPUT1"
+    data_type: {dtype}
+    dims: [ {dim} ]
+  }}
+]
+instance_group [ {{ kind: KIND_CPU }} ]
+'''.format(dtype=tensor_dtype, dim_len=len(tensor_shape),
+            batch_size=1024, dim=tu.shape_to_dims_str(tensor_shape))
+
+    try:
+        os.makedirs(config_dir)
+    except OSError as ex:
+        pass # ignore existing dir
+
+    with open(config_dir + "/config.pbtxt", "w") as cfile:
+        cfile.write(config)
+
+if __name__ == '__main__':
+    import argparse
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--models_dir', type=str, required=True,
+                        help='Top-level model directory')
+    FLAGS, unparsed = parser.parse_known_args()
+
+    # Create utility models used in ensemble
+    # nop (only creates model config, should add model file before use)
+    model_dtypes = ["TYPE_BOOL", "TYPE_STRING"]
+    for s in [8, 16, 32, 64]:
+      for t in ["INT", "UINT", "FP"]:
+        if t == "FP" and s == 8:
+          continue
+        model_dtypes.append("TYPE_{}{}".format(t, s))
+
+    for model_dtype in model_dtypes:
```

Comment:
How come this variable-size is done here and not in gen_qa_models.py like you do for the other models?